### PR TITLE
 Improve the testgrid extension architecture

### PR DESCRIPTION
**Pull request by kasunbg**
## Purpose
Details can be found in #497 and #498.

## Goals
Details can be found in #497 and #498.

## Approach
Details can be found in #497 and #498.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java [hunk 1]
```diff
@@ -48,13 +56,15 @@
...
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
 import java.util.Set;
 
 import static org.wso2.testgrid.common.TestGridConstants.PRODUCT_TEST_PLANS_DIR;
 
+//import org.wso2.testgrid.common.config.TestPlan;
```
Comment (pasindujw) [inline core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java]: commented line of code.

File: core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java [hunk 2]
```diff
@@ -57,6 +57,8 @@
 import java.util.List;
 import java.util.Optional;
 
+//import org.wso2.testgrid.common.config.TestPlan;
```
Comment (pasindujw) [inline core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java]: Commented line of code.

File: core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java (hunk 1 above)
Comment (kasunbg) [reply core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java]: Addressed all.

File: core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java [hunk 3]
```diff
@@ -83,63 +86,71 @@ public void runTestPlan(TestPlan testPlan, Infrastructure infrastructure) throws
...
-            testPlan.setDeployment(deployment);
+            InfrastructureProvider infrastructureProvider = InfrastructureProviderFactory
+                    .getInfrastructureProvider(infrastructureConfig);
+            infrastructureProvider.init();
+            InfrastructureProvisionResult provisionResult = infrastructureProvider
+                    .provision(infrastructureConfig, testPlan.getInfraRepoDir());
+
+            provisionResult.setName(infrastructureConfig.getProvisioners().get(0).getName());
+            provisionResult.setDeploymentScriptsDir(Paths.get(testPlan.getDeploymentRepoDir()).toString());
+            //            testPlan.setDeployment(deployment);
```
Comment (pasindujw) [inline core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java]: commented line of code.

Comment (kasunbg) [reply core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java]: good catch. fixed!

**Pull request merged**